driver/src/driver/scheduler/system.rs

```diff
@@ -279,61 +295,38 @@ mod tests {
         );
     }
 
+    // Allows observing real behavior by looking at the log output.
```

Comment:
nit: Could you make this comment a bit more descriptive, maybe include some expected output. The reason is when I read it I understood that log messages were being intercepted somehow and used to determine if the test succeeded or not.